Sources/NIOTransportServices/NIOTSConnectionBootstrap.swift

```diff
@@ -30,20 +51,20 @@ public final class NIOTSConnectionBootstrap {
 
     /// Create a `NIOTSConnectionBootstrap` on the `EventLoopGroup` `group`.
     ///
-    /// This initializer only exists to be more in-line with the NIO core bootstraps, in that they
-    /// may be constructed with an `EventLoopGroup` and by extenstion an `EventLoop`. As such an
-    /// existing `NIOTSEventLoop` may be used to initialize this bootstrap. Where possible the
-    /// initializers accepting `NIOTSEventLoopGroup` should be used instead to avoid the wrong
-    /// type being used.
-    ///
-    /// Note that the "real" solution is described in https://github.com/apple/swift-nio/issues/674.
+    /// The `EventLoopGroup` `group` must be compatible, otherwise the program will crash. `NIOTSConnectionBootstrap` is
+    /// compatible only with `NIOTSEventLoopGroup` as well as the `EventLoop`s returned by
+    /// `NIOTSEventLoopGroup.next`. See `init(validatingGroup:)` for a fallible initializer for
+    /// situations where it's impossible to tell ahead of time if the `EventLoopGroup` is compatible or not.
     ///
     /// - parameters:
     ///     - group: The `EventLoopGroup` to use.
-    public init(group: EventLoopGroup) {
-        self.group = group
+    public convenience init(group: EventLoopGroup) {
+        guard NIOTSBootstraps.isCompatible(group: group) else {
+            preconditionFailure("NIOTSConnectionBootstrap is only compatible with MultiThreadedEventLoopGroup and " +
+                                "SelectableEventLoop. You tried constructing one with \(group) which is incompatible.")
```

Comment:
The compatible types need updating here

Reply:
thanks! I did so much grepping but forgot that one :P

Reply:
fixed